### PR TITLE
Create RestClient using keystore, add comm-attr to user, cleanup

### PR DESCRIPTION
**Pull request by skkosuri-amzn**
*Issue #, if available:*
Create RestClient using keystore, add comm-attr to user, cleanup
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/amazon/opendistroforelasticsearch/commons/InjectSecurityTest.java [hunk 1]
```diff
@@ -77,7 +79,7 @@ public void testInjectUser() {
         assertEquals("plugin", threadContext.getTransient("ctx.name"));
 
         try (InjectSecurity helper = new InjectSecurity("test-name", settings, threadContext)) {
-            helper.inject("joe", "ops-role,logs-role");
+            helper.inject("joe", Arrays.asList("ops-role,logs-role"));
```
Comment (lezzago) [inline src/test/java/com/amazon/opendistroforelasticsearch/commons/InjectSecurityTest.java:82]: This is considered as 1 element string. Do you want 2 elements in the array?

File: src/test/java/com/amazon/opendistroforelasticsearch/commons/InjectSecurityTest.java [hunk 2]
```diff
@@ -51,7 +53,7 @@ public void testInjectRoles() {
...
```
Comment (lezzago) [inline src/test/java/com/amazon/opendistroforelasticsearch/commons/InjectSecurityTest.java:56]: This is considered as 1 element string. Do you want 2 elements in the array?

File: src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/IntegrationTests.java [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    @Test
+    public void testCreateRestClientWithUser() throws Exception {
+        RestClient client = new SecureRestClientBuilder("localhost", 9200, true, "admin", "admin").build();
+        Response response = client.performRequest(createSampleRequest());
+        String responseBody = EntityUtils.toString(response.getEntity());
+        assertEquals(200, response.getStatusLine().getStatusCode());
+        assertTrue(responseBody.contains("user_name\":\"admin"));
+
+        client.close();
```
Comment (lezzago) [inline src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/IntegrationTests.java:55]: Why are you only closing the client for this specific test?

File: src/main/java/com/amazon/opendistroforelasticsearch/commons/rest/SecureRestClientBuilder.java [hunk 4]
```diff
@@ -65,38 +69,66 @@
...
-        this.trustCert = pemFile;
-        this.configPath = configPath;
+        this.user = user;
+        this.passwd = passWord;
+        this.settings = Settings.EMPTY;
+        this.configPath = null;
+
+        if (Strings.isNullOrEmpty(user) || Strings.isNullOrEmpty(passwd)) {
+            throw new IllegalArgumentException("Invalid user or password");
+        }
```
[the remark this answers is not in the document]
Comment (skkosuri-amzn) [reply src/main/java/com/amazon/opendistroforelasticsearch/commons/rest/SecureRestClientBuilder.java:104]: +1

File: src/test/java/com/amazon/opendistroforelasticsearch/commons/InjectSecurityTest.java (hunk 1 above)
Comment (skkosuri-amzn) [reply src/test/java/com/amazon/opendistroforelasticsearch/commons/InjectSecurityTest.java:82]: good catch, fixed.

File: src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/IntegrationTests.java (hunk 3 above)
Comment (skkosuri-amzn) [reply src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/IntegrationTests.java:55]: Added to other tests also.

File: src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/SecureRestClientBuilderTest.java [hunk 5]
```diff
@@ -53,7 +53,6 @@ public void testHttpsRestClient() throws Exception {
         restClient.close();
     }
 
-    @Test(expected = ElasticsearchException.class)
     public void testMissingPem() throws Exception {
```
Comment (lezzago) [inline src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/SecureRestClientBuilderTest.java:56]: Why are you removing this test? If this is no longer valid, please remove the entire test.

Comment (skkosuri-amzn) [reply src/test/java/com/amazon/opendistroforelasticsearch/commons/rest/SecureRestClientBuilderTest.java:56]: Test is still valid. If the pem is missing, it falls back to selftrusted and test case succeeds.

**Review by lezzago — APPROVED** [review]

**Pull request merged**